src/FredData.jl

```diff
@@ -67,20 +67,29 @@ mutable struct Fred
     end
 end
 Fred(key::AbstractString) = Fred(key, DEFAULT_API_URL)
-function Fred()
-    key = if KEY_ENV_NAME in keys(ENV)
+
+function fred_key()
+    if KEY_ENV_NAME in keys(ENV)
         ENV[KEY_ENV_NAME]
     elseif isfile(joinpath(homedir(), KEY_FILE_NAME))
         open(joinpath(homedir(), KEY_FILE_NAME), "r") do file
             rstrip(read(file, String))
         end
     else
+        nothing
+    end
+end
+
+function Fred()
+    key = fred_key()
+    if key === nothing
```

Comment:
I don't see an advantage to introducing `nothing` values.
1. the change in logic has not caused any behavior in the FredData.jl public API
2. users who want to specifically call `load_fred_data` continue to have to test for the sentinel value `nothing` rather than having the method fail fast

I think we can safely make the body as follows:
```
function Fred()
    key = load_fred_key()
    println("API key loaded.")
    return Fred(key)
end
```

where the error is now raised by `load_fred_key`

Reply:
I see your point, that someone looking for the key can't rely on it coming through. I'll set up a boolean check.